app/models/piece.rb

```diff
@@ -32,4 +32,24 @@ def set_image
     color ? color_string = "white" : color_string = "black"
     self.image_name ||= "#{color_string}-#{piece_type.downcase}.png"
   end
+
+  def move_to!(new_x, new_y)
+    @target = Piece.where(:x_position => new_x).where(:y_position => new_y).take
+    if @target.nil?
+      update_attributes(:x_position => new_x, :y_position => new_y)
+    else
+      if color != @target.color
+        capture(new_x, new_y)
+      else
+        fail "Invalid move!"
```

Comment:
Might want to be slightly more descriptive for this error message and say something along the lines of how you can't move into a space occupied by another one of your own pieces so the user knows what they did wrong.